wandb/sdk/lib/credentials.py

```diff
@@ -0,0 +1,85 @@
+import json
+import os
+from datetime import datetime, timedelta
+from typing import Optional
+
+import requests.utils
+
+from wandb.errors import AuthenticationError
+
+DEFAULT_WANDB_CREDENTIALS_FILE = os.path.expanduser("~/.config/wandb/credentials.json")
+
+_expires_at_fmt = "%Y-%m-%d %H:%M:%S"
+
+
+def access_token(
+    base_url: str, token_file: str, credentials_file: str
+) -> Optional[str]:
+    if not token_file:
+        return None
+
+    if not os.path.exists(credentials_file):
+        write_credentials_file(base_url, token_file, credentials_file)
+
+    data = fetch_credentials(base_url, token_file, credentials_file)
+    return data["access_token"]
+
+
+def write_credentials_file(base_url: str, token_file: str, credentials_file: str):
+    credentials = create_access_token(base_url, token_file)
+    data = {"credentials": {base_url: credentials}}
+    with open(credentials_file, "w") as file:
+        json.dump(data, file, indent=4)
+
+        # Set file permissions to be read/write by the owner only
+        os.chmod(credentials_file, 0o600)
+
+
+def fetch_credentials(base_url: str, token_file: str, credentials_file: str) -> dict:
+    with open(credentials_file) as file:
+        data = json.load(file)
+        creds = data["credentials"][base_url]
```

Comment:
do you need to verify that the content of the file is correct, could it be that the json be corrupted or some fields are missing, just asking if we should handle missing fields

Reply:
can't hurt, handled missing fields